### ISSUE TITLE
Zooming in browser and saving changes the saved file size

### ISSUE THREAD

**Issue by wdrach** [processing/p5.js#5892]
### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [X] Utilities
- [ ] WebGL
- [ ] Build Process
- [ ] Unit Testing
- [ ] Internalization
- [ ] Friendly Errors
- [ ] Other (specify if possible)

### p5.js version

1.4.1

### Web browser and version

107.0.5304.121

### Operating System

MacOSX (ARM)

### Steps:
1. Open any p5.js sketch
2. Right click save the image, or use `save()` to save the image
3. Zoom way in on your browser
4. Re-run the script
5. Right click save the image again

Despite the canvas size not changing, the file from the zoomed in version will have a massively higher resolution. The createCanvas documentation even mentions that the width and height of the canvas is in pixels, so I don't think this is intended!

### Snippet:
This works on the default sketch in the p5 web editor. You don't even have to do anything special!

```javascript
function setup() {
  createCanvas(400, 400);
}

function draw() {
  background(220);
}
```

**Comment by davepagurek** [comment]
A short answer: to prevent this behaviour, after `createCanvas`, call [`pixelDensity(1)`.](https://p5js.org/reference/#/p5/pixelDensity)

Longer answer:

This is actually intended behaviour! There are, confusingly, two different concepts in web development both called "pixels": *physical* pixels (actual units of color data on your canvas) and *CSS* pixels (the units used for sizing things on a webpage.) There's a browser property called `devicePixelRatio` which, [according to MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio), describes:

> the ratio of pixel sizes: the size of one CSS pixel to the size of one physical pixel. In simpler terms, this tells the browser how many of the screen's actual pixels should be used to draw a single CSS pixel.

When you use a high dpi screen like a Macbook monitor, or when you zoom in, `devicePixelRatio` increases in order to not render web pages pixelated. P5 also uses more pixels in the canvas by default too, but you can tell p5 exactly how many physical pixels to CSS pixels to use with `pixelDensity(n)` rather than defaulting to `devicePixelRatio`. If you call `pixelDensity(1)`, things will work as expected.

**Comment by wdrach** [comment]
@davepagurek thanks so much for the detailed answer! I figured it was something silly like this, glad this issue is out here for anyone else experiencing this problem. I am indeed on a higher density device, so it was doubly messy.
